src/pybind/mgr/dashboard/tools.py

```diff
@@ -42,6 +54,9 @@ def _setup(self):
     def request_begin(self):
         req = cherrypy.request
         user = JwtManager.get_username()
+        if user is not None:
```

Comment:
to be pythonish it is better to do: `if user:` instead of `if user is not None:`, however I'm not sure you need this check, for example, later in `logger.debug` there is user used, but this code will fail when user==None in python3.

Reply:
Thanks, sorry for the late reply. I didn't see the comment.

Seems OK to pass None to a logging call, I tested with
```
import logging
logging.basicConfig(level=logging.DEBUG)
logger = logging.getLogger(__name__)
logger.debug('%s', None)
```
The check is needed because sometimes `JwtManager.get_username()` returns None, and ensure_str will fails on it.
https://github.com/ceph/ceph/blob/nautilus/src/pybind/mgr/dashboard/services/auth.py#L79-L81

Reply:
Apparently it works with debug, but it does not work always with format, see:
```
Python 3.7.7 (default, Mar 10 2020, 15:43:33) 
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> x=None
>>> "%s" % x # works
'None'
>>> "{x}".format(x=x) # works
'None'
>>> "{x:8s}".format(x=x) # does not work
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported format string passed to NoneType.__format__
>>> "{x:8s}".format(x='x') # work
'x       '
```

Reply:
String formatting in logging calls needs to be lazy. So we are not using `'{}'.format(val)` here.

Reply:
really? what is this then:
```
        if Settings.AUDIT_API_ENABLED and req.method not in ['GET']:
            url = build_url(req.remote.ip, scheme=req.scheme,
                            port=req.remote.port)
            msg = '[DASHBOARD] from=\'{}\' path=\'{}\' method=\'{}\' ' \
                'user=\'{}\''.format(url, req.path_info, req.method, user)
```
I know it is not a big deal, I'm just saying that having:
```
user = JwtManager.get_username() or 'Undefined'
```
Will save from all those non-necessary checks.

Reply:
> really? what is this then:
> 
> ```
>         if Settings.AUDIT_API_ENABLED and req.method not in ['GET']:
>             url = build_url(req.remote.ip, scheme=req.scheme,
>                             port=req.remote.port)
>             msg = '[DASHBOARD] from=\'{}\' path=\'{}\' method=\'{}\' ' \
>                 'user=\'{}\''.format(url, req.path_info, req.method, user)
> ```
> 

As you can see it's already passed `if Settings.AUDIT_API_ENABLED` check, so being lazy here is not necessary.